minimal.Dockerfile

```diff
@@ -0,0 +1,61 @@
+# Steps to build and push minimal SMARTS docker image
+# ```bash
+# $ export VERSION=v0.4.13
+# $ cd /path/to/SMARTS
+# $ docker build -t huaweinoah/smarts:$VERSION-minimal -f minimal.Dockerfile .
+# $ docker push huaweinoah/smarts:$VERSION-minimal
+# ```
+
+FROM ubuntu:bionic
+
```

Comment:
The removal of the lines to force non-interactive setup is painful.

```suggestion

ARG DEBIAN_FRONTEND=noninteractive

# Prevent tzdata from trying to be interactive
ENV TZ=Europe/Minsk
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

# http://bugs.python.org/issue19846
# > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
ENV LANG C.UTF-8

```

Reply:
This `noninteractive` command was removed to reduce the number of layers in the image. Instead, I have added the `-y` option wherever necessary to prevent interactivity. I did not face any interactivity during build, but if you faced any interactivity during docker build, please let me know so that we can add default options such as `-y` to the particular commands. 
```dockerfile
ARG DEBIAN_FRONTEND=noninteractive
```

I am not sure why these lines were included originally as they don't seem to have any effect on the code. Hence, I removed them.
```dockerfile
# Prevent tzdata from trying to be interactive
ENV TZ=Europe/Minsk
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

# http://bugs.python.org/issue19846
# > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
ENV LANG C.UTF-8
```

Reply:
OK, I got interactive commands for setting up the keyboard.

Reply:
Since I don't get the request to setup the keyboard, could you help narrow down the line which throws this interactive request? 

Please replace the first `RUN` command with the following to better identify the line of interest.

```dockerfile
RUN apt-get update --fix-missing 
RUN apt-get install -y software-properties-common
RUN add-apt-repository -y ppa:deadsnakes/ppa
RUN add-apt-repository -y ppa:sumo/stable
RUN apt-get update 
RUN apt-get install -y
        libsm6 \
        libspatialindex-dev \
        libxext6 \
        libxrender-dev \
        python3.7 \
        python3.7-dev \
        python3.7-venv \
        sumo \
        sumo-doc \
        sumo-tools \
        wget \
        x11-apps \
        xserver-xorg-video-dummy
RUN apt-get autoremove -y
RUN rm -rf /var/lib/apt/lists/* 
```